blazar_dashboard/content/leases/tables.py

```diff
@@ -136,3 +138,18 @@ class Meta(object):
             table_actions.insert(0, ViewDeviceReservationCalendar)
 
         row_actions = (UpdateLease, DeleteLease, )
+
+    def __init__(self, *args, **kwargs):
+        super(LeasesTable, self).__init__(*args, **kwargs)
+        user_id_column = next((c for c in self.get_columns() if c.name == "user_id"), None)
+        if user_id_column:
+            user_id_column.filters.append(lambda u: self.uid_to_user(u))
+
+    def uid_to_user(self, uid):
+        if not uid:
+            return None
+        try:
+            user = horizon_api.keystone.user_get(self.request, uid, admin=False)
```

Comment:
Perhaps at the very least these could be batched together? Though it will be bottlenecked by the serial keystone requests still.

Reply:
Added a cache decorator to the lookup function.